diffkemp/semdiff/pattern_config.py

```diff
@@ -0,0 +1,129 @@
+"""Configuration for difference code patterns."""
+from subprocess import check_call, CalledProcessError, DEVNULL
+import os
+import yaml
+
+
+class UnsupportedPatternError(ValueError):
+    """Indicates that a pattern file uses an unsupported format."""
+    pass
+
+
+class PatternConfig:
+    """
+    Configuration file containing difference code patterns and global pattern
+    comparison settings.
+    Difference patterns can be used during module comparison to prevent
+    the reporting of known and desired semantic differences.
+    """
+    def __init__(self, path=None, patterns_path=None):
+        """
+        Create a new difference pattern configuration.
+        :param path: Path to the configuration source.
+        """
+        self.path = path
+        self.patterns_path = patterns_path
+        self.pattern_files = set()
+
+        self.settings = {
+            "on_parse_failure": "ERROR"
+        }
+
+        self.llvm_only = True
+
+    @classmethod
+    def create_from_file(cls, path, patterns_path=None):
+        """
+        Creates a new difference pattern configuration from a single pattern
+        file or a configuration YAML.
+        :param path: Path to the file to load.
+        """
+        PatternConfig.raise_for_invalid_file(path)
+        config = cls(path, patterns_path)
+
+        # Process the configuration file based on its extension.
+        if PatternConfig.get_extension(path) not in ("ll"):
```

Comment:
Not necessarily. However, it feels cleaner as the function is used on line 82 as well.

Reply:
Ok then. It would be nice to have this outside of the `PatternConfig` class, but we don't have any "utils" file in the Python part, yet. Let's leave it as it is.